cmd/ketch/app.go

```diff
@@ -26,3 +31,80 @@ func newAppCmd(cfg config, out io.Writer) *cobra.Command {
 	cmd.AddCommand(newAppExportCmd(cfg, out, appExport))
 	return cmd
 }
+
+func appState(pods []apiv1.Pod) (state string) {
+	state = "unknown"
+	// If there's no pods for this app, then app in `created` state
+	if len(pods) == 0 {
+		state = strings.ToLower(string(ketchv1.AppCreated))
+		return state
+	}
+
+	// If app has pods, then we build app status based on pods statuses
+	var deploying, running, errored, succeeded int
+	for _, pod := range pods {
+		switch podState(pod) {
+		case ketchv1.PodRunning:
+			running++
+		case ketchv1.PodDeploying:
+			deploying++
+		case ketchv1.PodError:
+			errored++
+		case ketchv1.PodSucceeded:
+			succeeded++
+		}
+	}
+
+	var parts []string
+	if deploying > 0 {
+		parts = append(parts, fmt.Sprintf(`%d %s`, deploying, ketchv1.PodDeploying))
+	}
+	if running > 0 {
+		parts = append(parts, fmt.Sprintf(`%d %s`, running, ketchv1.PodRunning))
+	}
+	if errored > 0 {
+		parts = append(parts, fmt.Sprintf(`%d %s`, errored, ketchv1.PodError))
+	}
+	if succeeded > 0 {
+		parts = append(parts, fmt.Sprintf(`%d %s`, succeeded, ketchv1.PodSucceeded))
+	}
+
+	return strings.Join(parts, ", ")
+}
+
+var (
+	containerStateCrashLoopBackOff = "CrashLoopBackOff"
+	containerStateCompleted        = "Completed"
+)
+
+func podState(pod apiv1.Pod) ketchv1.PodState {
+	switch pod.Status.Phase {
+	case apiv1.PodPending:
+		return ketchv1.PodDeploying
+	case apiv1.PodRunning:
+		// In some cases, we have to override the status obtained by using `pod.Status.Phase` because it could be
+		// `running` state while the actual container in `CrashLoopBackOff` so, actual state of the pod is `error`,
+		// and so on.
+		for _, c := range pod.Status.ContainerStatuses {
+			// If any of the pod containers in the `CrashLoopBackOff` state we must return the `error` state
+			if c.State.Waiting != nil {
+				if c.State.Waiting.Reason == containerStateCrashLoopBackOff {
+					return ketchv1.PodError
+				}
+			}
+			// If any of the pod containers in the `Completed` state we must return the `succeeded` state
+			if c.State.Terminated != nil {
+				if c.State.Terminated.Reason == containerStateCompleted {
```

Comment:
what if there are two containers: the first one has `completed` state and the second has `containerStateCrashLoopBackOff` state?

Reply:
then first happened error will take precedence. If we will want to improve that logic - let's do that then.